cpp/ql/lib/semmle/code/cpp/security/BufferAccess.qll

```diff
@@ -29,7 +29,20 @@ abstract class BufferAccess extends Expr {
    */
   abstract Expr getBuffer(string bufferDesc, int accessType);
 
-  abstract int getSize();
+  /**
+   * Gets the expression that represents the size of the buffer access. The
+   * actual size is the value of this expression multiplied by the result of
+   * `getSizeMult()`, in bytes.
+   */
```

Comment:
At the moment, this isn't true for `ArrayExprBA`, see my comment there.

Reply:
How about just adding the word "typically" so it's clear there is an exception:
```
 /**
   * Gets the expression that represents the size of the buffer access. The
   * actual size is typically the value of this expression multiplied by the result of
   * `getSizeMult()`, in bytes.
   */
```